scanner.go

```diff
@@ -30,15 +36,40 @@ type scanner struct {
 	// curRegionScannerID is the id of scanner on current region
 	curRegionScannerID uint64
 	// startRow is the start row in the current region
-	startRow []byte
-	results  []*pb.Result
-	closed   bool
+	startRow    []byte
+	results     []*pb.Result
+	closed      bool
+	scanMetrics map[string]int64
+	sync.RWMutex
 }
 
 func (s *scanner) fetch() ([]*pb.Result, error) {
 	// keep looping until we have error, some non-empty result or until close
 	for {
 		resp, region, err := s.request()
+		if s.rpc.TrackScanMetrics() && resp.ScanMetrics != nil {
+			metrics := resp.ScanMetrics.GetMetrics()
+			s.Lock()
+			for _, m := range metrics {
```

Comment:
The for-loop can be replaced with just:
```go
for _, m := range metrics {
    s.scanMetrics[m.GetName()] += m.GetValue()
}
```
This way:
- no dependency between HBase and gohbase version
- `+=` on a map works since doing a map access on a key that doesn't already exist return the default value which for uint64 is 0